src/imagery/i.eodag/i.eodag.py

```diff
@@ -300,82 +396,199 @@ def no_fallback_search(search_parameters, provider):
+    minimum_overlap = kwargs["minimum_overlap"]
+    cloud_cover = kwargs["clouds"]
+    start_date = kwargs["start"]
+    end_date = kwargs["end"]
+
+    # If neither a geometry is provided as a paraemter
+    # nor a vector map is provided through "options",
+    # then none of the geometry filtering will take place.
+    if geometry is None and (area_relation is not None or minimum_overlap is not None):
+        geometry = get_aoi(kwargs["map"])
+    gs.verbose(_("Filtering results..."))
+
+    if area_relation or minimum_overlap:
+        # Product's geometry intersects with AOI
+        if area_relation == "Intersects":
+            search_result = search_result.filter_overlap(
+                geometry=geometry, intersects=True
+            )
+        # Product's geometry contains the AOI
+        elif area_relation == "Contains":
+            search_result = search_result.filter_overlap(
+                geometry=geometry, contains=True
+            )
+        # Product's geometry is within the AOI
+        elif area_relation == "IsWithin":
+            search_result = search_result.filter_overlap(geometry=geometry, within=True)
+        # Percentage of area covered from the AOI by the product's geometry
+        elif minimum_overlap:
+            search_result = search_result.filter_overlap(
+                geometry=geometry, minimum_overlap=int(minimum_overlap)
+            )
 
-def apply_filters(search_result):
-    filtered_result = []
-    for product in search_result:
-        valid = True
-        if (
-            options["clouds"]
-            and "cloudCover" in product.properties
-            and product.properties["cloudCover"] is not None
-            and product.properties["cloudCover"] > int(options["clouds"])
-        ):
-            valid = False
-        if valid:
-            filtered_result.append(product)
-    return filtered_result
+    if cloud_cover:
+        search_result = search_result.filter_property(
+            operator="le", cloudCover=int(cloud_cover)
+        )
+
+    search_result = search_result.filter_date(start=start_date, end=end_date)
+    search_result = remove_duplicates(search_result)
+
+    postfilter_count = len(search_result)
+    gs.verbose(
+        _("{} product(s) filtered out.".format(prefilter_count - postfilter_count))
+    )
+
+    return search_result
+
+
+def sort_result(search_result):
+    """Sorts search results according to options['sort'] and options['order']
+
+    options['sort'] parameters and options['order'] are matched correspondingly.
+    If options['order'] parameters are not suffcient,
+    'asc' will be used by default.
+
+    :param search_result: EO products to be sorted
+    :type search_result: class'eodag.api.search_result.SearchResult'
+
+    :return: Sorted EO products
+    :rtype: class:'eodag.api.search_result.SearchResult'
+    """
+    gs.verbose(_("Sorting..."))
+
+    sort_keys = options["sort"].split(",")
+    sort_order = options["order"].split(",")
```

Comment:
Maybe we can limit the `order` option to one value, either "asc" or "desc", that is how it is implemented in i.sentinel.download and i.landsat.download, and also because the GUI, AFAIK, can't have multiple values and preset limiting options at the same time, and it just allows the user to choose only one value from the drop-down once...

Reply:
Did that here ebc988487906cb2346efba999c12c71fc3770a9c.